File: src/test/java/edu/tallerjava/persistencia/UsuarioTest.java
package edu.tallerjava.persistencia;

import edu.tallerjava.modelo.Category;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.annotation.Rollback;
import org.springframework.test.context.junit4.SpringRunner;
import org.springframework.transaction.annotation.Transactional;

import javax.persistence.EntityManager;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(SpringRunner.class)
@SpringBootTest
public class UsuarioTest {

    @Autowired
    private EntityManager entityManager;

    @Test @Transactional @Rollback
    public void save(){
        Category juan = new Category();
        juan.setName("juan");

        entityManager.persist(juan);
        assertThat(entityManager.find(Category.class, juan.getId())).isNotNull();
    }
}

File: src/test/java/edu/tallerjava/domain/UsuarioTest.kt
package edu.tallerjava.domain

import org.assertj.core.api.Assertions
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.annotation.Rollback
import org.springframework.test.context.junit4.SpringRunner
import org.springframework.transaction.annotation.Transactional
import javax.persistence.EntityManager

@RunWith(SpringRunner::class)
@SpringBootTest
open class UsuarioTest {
    @Autowired
    private lateinit var entityManager: EntityManager

    @Test
    @Transactional
    @Rollback
    open fun save() {
        val juan = Usuario()
        juan.nombre = "juan"
        entityManager.persist(juan)
        Assertions.assertThat(entityManager.find(Usuario::class.java, juan.id)).isNotNull
    }
}